alot/ui.py

```diff
@@ -664,3 +672,37 @@ def handle_signal(self, signum, frame):
             if isinstance(self.current_buffer, SearchBuffer):
                 self.current_buffer.rebuild()
                 self.update()
+
+    def cleanup(self):
+        """Do the final clean up before shutting down."""
+        self._save_history_to_file(self.commandprompthistory, self._cache)
+
+    @staticmethod
+    def _load_history_from_file(path):
+        """Load a history list from a file and split it into lines.
+
+        :param path: the path to the file that should be loaded
+        :type path: str
+        :returns: a list of history items (the lines of the file)
+        :rtype: list(str)
+
+        """
+        if os.path.exists(path):
+            return [line.rstrip('\n') for line in open(path).readlines()]
+
+    @staticmethod
+    def _save_history_to_file(history, path):
+        """Save a history list to a file for later loading (possibly in another
+        session).
+
+        :param history: the history list to save
+        :type history: list(str)
+        :param path: the path to the file where to save the history
+        :type path: str
+        :returns: None
+
+        """
+        directory = os.path.dirname(path)
+        if not os.path.exists(directory):
+            os.makedirs(directory)
+        open(path, 'w').write('\n'.join(history))
```

Comment:
This will leak a file descriptor until the gc runs. Use a context manager instead:
```python
with open(path, 'w') as f:
    for l in history:
        f.write(l)
        f.write('\n')
```

This other advantage of this approach is that it avoids building a (potentially) large string in memory before writing it.